### PR TITLE
Fix: .uwu incorrectly transforms emojis in their raw text form

### LINKED ISSUE #1072: .uwu command incorrectly transforms emojis in their raw text form
## Description
The current .uwu feature does not handle or preserve external emojis

### Example 1
Example

### Example 2
Example_2

## Expected Behaviour
Either ignoring external emojis, preserving them in place (may not be possible), or transforming them into emoticons instead of the normal text-transform rules. In my opinion, the emoticon replacement is most suitable.

## Possible Solutions
- Detect external emojis by checking for the emoji ID within the bot's known emoji list. 
- For custom emojis, the bot would not be able to send the emojis back, such, escape this from text-transform rules, and instead replacing the custom emoji with a random emoticon from the emoticon list.

## Would you like to implement a fix?

- [x] I'd like to implement the bug fix
- [ ] Anyone can implement the bug fix

### PR DESCRIPTION
**Pull request by ionite34**
## Relevant Issues
Fixes #1072

## Description
6db84b38568d3f53c1b1bec4c66c1c8f2ee7b1ac
- Added pattern `REGEX_EMOJI` to capture emoji markdowns
- Implemented `Emoji` data class for emoji information and displayability inferencing
- Added function `_ext_emoji_replace` to replace external undisplayable emojis with random emoticons
- Added call to `_ext_emoji_replace` at the end of `_uwuify`

## Fixes / formatting
202d53b3f7ebe4d6041a2e5a4c1419bc16b27e11
- Added type checking annotations for `fun_cog` using `typing.TYPE_CHECKING`

21194a8817ee34229a211f6f5aae8978d054a884
- Replaced a redundant single-letter regex character class with the character itself

## Example
- Built-in emojis and emojis the bot is able to display (due to sharing a server) are returned normally.
- For external undisplayable emojis, the bot correctly replaces it with a random emoticon from the `EMOJIS` list.

Example

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read and agree to the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

File: bot/exts/fun/uwu.py [hunk 1]
```diff
@@ -53,6 +57,32 @@
 REGEX_NYA = re.compile(r"n([aeou][^aeiou])")
 SUBSTITUTE_NYA = r"ny\1"
 
+REGEX_EMOJI = re.compile(r"<(a?)?:(\w+):(\d{18})>?", re.ASCII)
```
Comment (ChrisLovering) [inline bot/exts/fun/uwu.py:60]:
```suggestion
REGEX_EMOJI = re.compile(r"<(a)?:(\w+?):(\d{15,21}?)>", re.ASCII)
```
The `?` inside the first match (a?) is unneeded as the is another ? outside the match block, doing the same thing.

The ID for the emoji is a snowflake, so it's actually valid between 15 and 21 characters.

File: bot/exts/fun/uwu.py [hunk 2]
```diff
@@ -96,6 +126,26 @@ def _emoji_replace(self, match: re.Match, strength: float = 0.0) -> str:
             return f" {random.choice(EMOJIS)} "
         return match_string
 
+    def _ext_emoji_replace(self, input_string: str) -> str:
+        """Replaces external emojis with emoticons."""
```
Comment (ChrisLovering) [inline bot/exts/fun/uwu.py:130]:
```suggestion
        """Replaces any emoji the bot cannot send in input_text with a random emoticons."""
```

**Review by ChrisLovering — CHANGES_REQUESTED** [review]
Just a few small comments, looks good otherwise.

I also really appreciate you including screenshots of it in action in your PR! :)

**Review by ChrisLovering — APPROVED** [review]

**Review by wookie184 — APPROVED** [review]
I don't have nitro so there was a limit to what I could test, but this seems good, thanks!

**Pull request merged**